### ISSUE TITLE
Usage of deprecated methods - x509.EncryptPEMBlock(), x509.IsEncryptedPEMBlock(), x509.DecryptPEMBlock()

### ISSUE THREAD

**Issue by achuchev** [Venafi/vcert#186]
**PROBLEM SUMMARY**
golangci-lint complains that in several vCert methods deprecated functions from x509 package are used but they are deprecated since Go 1.16.

**STEPS TO REPRODUCE**
1. Make sure that you have Go 1.16 or above
2. Execute `make test`

**EXPECTED RESULTS**
Tests to be successfully executed.

**ACTUAL RESULTS**
`golangci-lint` will fail with several errors. One example is below
```
pkg/certificate/certificate.go:542:10: SA1019: x509.EncryptPEMBlock has been deprecated since Go 1.16 because it shouldn't be used: Legacy PEM encryption as specified in RFC 1423 is insecure by design. Since it does not authenticate the ciphertext, it is vulnerable to padding oracle attacks that can let an attacker recover the plaintext. (staticcheck)
		return x509.EncryptPEMBlock(rand.Reader, "RSA PRIVATE KEY", x509.MarshalPKCS1PrivateKey(k), password, x509.PEMCipherAES256)
```

**ENVIRONMENT DETAILS**
Notting special, just Go 1.17.

**COMMENTS/WORKAROUNDS**
I have temporally disabled lint by annotating the problematic vCert methods with`//nolint`.

Example:
```
//nolint
func (o *Output) AsJKS(c *Config) ([]byte, error) {}
```

**Comment by tr1ck3r** [comment]
This is technically not a bug because VCert is currently built on Golang 1.15 but it is technical debt that will need to be addressed when we move it to Golang 1.16 or higher.  Having support for the deprecated PKCS#1 key format has been valuable to customers with legacy devices that do not support PKCS#8 but such cases are becoming increasingly rare so we may be able to address this by adding support for PKCS#8 and dropping support for PKCS#1.  If we need to continue supporting PKCS#1 we would do so by porting the native Golang code over to our project which would eliminate the deprecation warning.  But right now I'm leaning toward ending support for PKCS#1.

**Comment by achuchev** [comment]
Makes sense.

**Comment by tr1ck3r** [comment]
@angelmoo please address this by doing the following:

1. Change the default output format for PEM private keys from PKCS#1 to PKCS#8.  I believe this will require the use of the https://github.com/youmark/pkcs8 module to allow for encrypted private keys.  
2. Clone the deprecated methods as needed from the Go x509 package (e.g. DecryptPEMBlock, EncryptPEMBlock, IsEncryptedPEMBlock) so VCert can continue to support the PKCS#1 output format while avoiding any deprecation warnings from the Go builder.  Continued support is based on customer feedback and the general objective of VCert to eliminate the need to use other crypto tools like `openssl`.
3. Introduce a new value "legacy-pem" to the `--format` parameter that, when specified, results in private keys being output in PKCS#1 format the same as they are currently (password encrypted or clear based on whether `--no-prompt` was specified)

**Comment by tr1ck3r** [comment]
This was addressed by #198 and first appeared in the [v4.17.0](https://github.com/Venafi/vcert/releases/tag/v4.17.0) release.
